jpo-ode-svcs/src/main/java/us/dot/its/jpo/ode/services/asn1/message/Asn1DecodeMessageJSON.java

```diff
@@ -47,74 +50,118 @@ public Asn1DecodeMessageJSON(OdeProperties odeProps) {
-						
+
+						publishEncodedMessageToAsn1Decoder(odeData);
+					}
+					
+				} else if (key != null && key.toString().equals(SPATContentType)) {
+					/**
+					 * process consumed data { "SpatMessageContent": [{ "metadata": { "utctimestamp:
+					 * "2020-11-30T23:45:24.913657Z" }
+					 * "payload":"00131A604A380583702005837800080008100000040583705043002580"}]}
+					 */
+					
+					OdeSpatMetadata metadata = null;
+					JSONArray rawSPATJsonContentArray = rawJSONObject.getJSONArray(SPATContentType);
+					for (int i = 0; i < rawSPATJsonContentArray.length(); i++) {
+						JSONObject rawSPATJsonContent = (JSONObject) rawSPATJsonContentArray.get(i);
+						String encodedPayload = rawSPATJsonContent.get("payload").toString();
+						JSONObject rawmetadata = (JSONObject) rawSPATJsonContent.get("metadata");
+
+						// construct payload
+						payload = new OdeAsn1Payload(new OdeHexByteArray(encodedPayload));
+
+						// construct metadata
+						metadata = new OdeSpatMetadata(payload);
+						metadata.setOdeReceivedAt(rawmetadata.getString("utctimestamp"));
+						metadata.setRecordType(RecordType.spatTx);
+						metadata.setSecurityResultCode(SecurityResultCode.success);
+
+						// construct metadata: receivedMessageDetails
+						ReceivedMessageDetails receivedMessageDetails = new ReceivedMessageDetails();
+						receivedMessageDetails.setRxSource(RxSource.NA);
+
+						// construct metadata: locationData
+						OdeLogMsgMetadataLocation locationData = new OdeLogMsgMetadataLocation();
+						receivedMessageDetails.setLocationData(locationData);
+
+						metadata.setReceivedMessageDetails(receivedMessageDetails);
+						metadata.setSpatSource(SpatSource.RV);
+
+						Asn1Encoding unsecuredDataEncoding = new Asn1Encoding("unsecuredData", "MessageFrame",
+								EncodingRule.UPER);
+						metadata.addEncoding(unsecuredDataEncoding);
+
+						// construct odeData
+						odeData = new OdeAsn1Data(metadata, payload);
+
 						publishEncodedMessageToAsn1Decoder(odeData);
 					}
 				} else {
 					logger.error("Error received null key from consumed message");
 				}
 			}
 
-			
+		} catch (
 
-		} catch (Exception e) {
+		Exception e) {
 			logger.error("Error publishing to Asn1DecoderInput: {}", e.getMessage());
 		}
 		return null;
 	}
 
-	private void publishEncodedMessageToAsn1Decoder(OdeData odeData){
+	private void publishEncodedMessageToAsn1Decoder(OdeData odeData) {
 		XmlUtils xmlUtils = new XmlUtils();
 		try {
 			logger.info("Sending encoded payload XML to ASN1 codec {}", xmlUtils.toXml(odeData));
-			codecPublisher.publish(xmlUtils.toXml(odeData),codecPublisher.getOdeProperties().getKafkaTopicAsn1DecoderInput());
+			codecPublisher.publish(xmlUtils.toXml(odeData),
+					codecPublisher.getOdeProperties().getKafkaTopicAsn1DecoderInput());
 		} catch (JsonProcessingException e) {
 			logger.info("Error sending encoded payload XML to ASN1 codec {}", e.getMessage());
 			e.printStackTrace();
 		}
-		
+
```

Comment:
@dan-du-car Didn't see any unit tests.  Sonar scan is reporting 0% coverage.  We should add unit tests for at least the new code if none exist.

Reply:
@snallamothu Currently all the unit test is set to be ignored. We have another existing PR addressing the unit tests.